src/util/OAuth2Util.js

```diff
@@ -31,14 +31,20 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
    * @param params - {idp: 'xxx'} for social auth
    *                 {sessionToken: 'xxx'} for okta idp
    */
-  util.getTokens = function (settings, params) {
+  util.getTokens = function (settings, params, controller) {
 
     function success(result) {
       settings.callGlobalSuccess(Enums.SUCCESS, result);
     }
 
     function error(error) {
-      settings.callGlobalError(new Errors.OAuthError(error.message));
+      // OKTA-104330- Handle error case where user is not assigned to OIDC client
+      if (error.errorCode == 'access_denied') {
+        controller.model.trigger('error', controller.model, {'responseJSON': error});
```

Comment:
Ah, I see. Yeah the ideal case would be that we actually return a promise here, and let RouterUtil handle it, but we can do that later since we'd have to fix the rest of this file at the same time. Okay, I'm okay with this change then.